lizard_connector/connector.py

```diff
@@ -208,18 +213,90 @@ def paginated(self):
         return bool(self.next_url)
 
 
-class Timeseries(Endpoint):
-    endpoint = 'timeseries'
-
-
-class Rasters(Endpoint):
-    endpoint = 'rasters'
-
-
-class Events(Endpoint):
-    endpoint = 'events'
+class Datatype(object):
+    """
+    Experimental class that forms the building block for data types.
 
+    Each data type contains a list of queries for each relevant lizard-api
+    endpoint. These are the only viable queries for each endpoint.
+    """
 
-# TODO:
-# class Assets(Endpoint):
-#     endpoint = ''
+    @property
+    def queries(self):
+        """
+        This property needs to be implemented for the class to function.
+        """
+        raise NotImplementedError
+
+    def list_endpoints(self):
+        """Lists all available endpoints for this datatype."""
+        return list(self.queries.keys())
+
+    def download(self, endpoint, *querydicts, **queries):
+        """Downloads a json as a dict from an endpoint."""
+        endpoint_ = Endpoint(endpoint)
+        return endpoint_.download(*querydicts, **queries)
+
+
+class Timeseries(Datatype):
+    queries = {
+        "timeseries": {
+            "in_bbox": functools.partial(
```

Comment:
Voelde eerlijk gezegd ook niet helemaal Pythonic. Het is een beperkte vorm van currying. Zoals ook in de docstrings aangegeven is dit een 'expirimenteel' deel. De vraag vanuit Arjen was om het aan te laten sluiten op de datatypes. Daar heb ik dus classes van gemaakt. Ik heb het gedaan omdat sommige queries hebben endpoint specifieke informatie nodig die je al kan invullen als je weet dat de query voor een bepaald enpoint geld. 
Hier praat ik nog graag even over door.